test/with_dummyserver/test_socketlevel.py

```diff
@@ -1442,14 +1442,14 @@ def _test_broken_header_parsing(self, headers):
 
     def test_header_without_name(self):
         self._test_broken_header_parsing([
-            b': Value\r\n',
-            b'Another: Header\r\n',
+            b': Value',
+            b'Another: Header',
```

Comment:
These `\r\n` changes don't make much sense to me. Should it also have switched to `b'\r\n'.join(headers)` up above?

Reply:
Okay, so the [initial commit in the master branch](https://github.com/urllib3/urllib3/commit/5c2cf76ddd2d42aa1906aef5eb08c6891a9b2614) had those extra \r\n, which was a mistake. This was fixed twice:

 * [in the v2 branch](https://github.com/urllib3/urllib3/commit/f220a99eb2a1b1bb3d9ae30b58904c318fce63db) by switching to `b''.join()`
 * [in the master branch (the commit I just merged)](https://github.com/urllib3/urllib3/commit/f4efcca559cc753f878e0a73e28dd6e795d0d052) by fixing the individual test cases and keeping `b'\r\n'.join()`.

So I just pushed a new commit that follows the approach of the master branch, to help future merges. This should make more sense!